src/OpenTelemetry.Exporter.Geneva/Metrics/OtlpProtobuf/OtlpProtobufSerializer.cs

```diff
@@ -0,0 +1,353 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.Exporter.Geneva;
+
+internal class OtlpProtobufSerializer
+{
+    private const int LengthAndTagSize = 4;
+
+    private readonly Dictionary<string, List<Metric>> scopeMetrics = new();
+
+    private int resourceMetricStartIndex;
+
+    private int scopeMetricsStartIndex;
+
+    private int metricStartIndex;
+
+    private int intrumentStartIndex;
+
+    private int metricPointStartIndex;
+
+    internal IMetricDataTransport MetricDataTransport;
+
+    public OtlpProtobufSerializer(IMetricDataTransport metricDataTransport)
+    {
+        this.MetricDataTransport = metricDataTransport;
+    }
+
+    internal void SerializeAndSendMetrics(byte[] buffer, Resource resource, in Batch<Metric> metricBatch)
+    {
+        foreach (var metric in metricBatch)
+        {
+            if (this.scopeMetrics.TryGetValue(metric.MeterName, out var metricList))
+            {
+                metricList.Add(metric);
+            }
+            else
+            {
+                var newList = new List<Metric>();
+                newList.Add(metric);
+                this.scopeMetrics[metric.MeterName] = newList;
+            }
+        }
+
+        // Serialize
+        this.SerializeResourceMetrics(buffer, resource);
+
+        this.ClearScopeMetrics();
+    }
+
+    internal void ClearScopeMetrics()
+    {
+        foreach (var list in this.scopeMetrics)
```

Comment:
It's not list, rather KeyValuePair, either use the correct type instead of var or rename list to something like entry.